java/dynamodb-tools/src/test/java/sleeper/dynamodb/tools/DynamoDBToolsTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.dynamodb.tools;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeDefinition;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.KeySchemaElement;
+import com.amazonaws.services.dynamodbv2.model.KeyType;
+import com.amazonaws.services.dynamodbv2.model.PutItemRequest;
+import com.amazonaws.services.dynamodbv2.model.ScalarAttributeType;
+import com.amazonaws.services.dynamodbv2.model.ScanRequest;
+import com.amazonaws.services.dynamodbv2.model.ScanResult;
+import org.junit.After;
+import org.junit.Test;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static sleeper.dynamodb.tools.DynamoDBUtils.initialiseTable;
+
+public class DynamoDBToolsTest extends DynamoDBTestBase {
+    private final static String TEST_KEY = "test-key";
+    private final static String TEST_VALUE = "test-value";
+    private final static String TEST_TABLE_NAME = "dynamodb-tools-test-table";
+
+    @After
+    public void tearDown() {
+        dynamoDBClient.deleteTable(TEST_TABLE_NAME);
+    }
+
+    @Test
+    public void shouldCreateRecordWithStringAttribute() {
+        // Given we have a table in dynamodb that accepts strings
+        createStringTable();
+        // When we create a record with string attributes
+        Map<String, AttributeValue> record = new DynamoDBRecordBuilder()
+                .string(TEST_KEY, UUID.randomUUID().toString())
+                .string(TEST_VALUE, "value").build();
+        // And save that record to dynamodb
+        dynamoDBClient.putItem(new PutItemRequest(TEST_TABLE_NAME, record));
+        // Then we should be able to retrieve the record from dynamodb
+        ScanResult result = dynamoDBClient.scan(new ScanRequest().withTableName(TEST_TABLE_NAME));
+        assertThat(result.getItems()).hasSize(1);
+        assertThat(result.getItems().get(0)).isEqualTo(record);
```

Comment:
We can use list assertions for each of these tests, ie. assertThat(result.getItems()).containsExactly(record).